plugin/pkg/shoot/validator/admission_test.go

```diff
@@ -293,9 +318,10 @@ var _ = Describe("validator", func() {
 				Expect(coreInformerFactory.Core().InternalVersion().CloudProfiles().Informer().GetStore().Add(&cloudProfile)).To(Succeed())
 				Expect(coreInformerFactory.Core().InternalVersion().Seeds().Informer().GetStore().Add(&seed)).To(Succeed())
 				Expect(coreInformerFactory.Core().InternalVersion().SecretBindings().Informer().GetStore().Add(&secretBinding)).To(Succeed())
+				auth.EXPECT().Authorize(ctx, authorizeAttributes).Return(authorizer.DecisionAllow, "", nil)
```

Comment:
Instead of repeating this expectation over and over again, you could add it in `JustBeforeEach`. For example:
```go
BeforeEach(func() {
	ctrl = gomock.NewController(GinkgoT())
	auth = nil
	// ...
})

JustBeforeEach(func() {
	if auth == nil {
		auth = mockauthorizer.NewMockAuthorizer(ctrl)
		auth.EXPECT().Authorize(ctx, gomock.Any()).Return(authorizer.DecisionAllow, "", nil).AnyTimes()
	}
	admissionHandler.SetAuthorizer(auth)
})

Context("setting spec.seedName", func() {
	BeforeEach(func() {
		auth = mockauthorizer.NewMockAuthorizer(ctrl)
	})
// ...
})
```

This will make the test more readable/maintainable and also this PR more reviewable.